src/nnet3/nnet-simple-component.cc

```diff
@@ -186,7 +187,111 @@ const BaseFloat NormalizeComponent::kNormFloor = pow(2.0, -66);
+  std::string tok; // TODO: remove back-compatibility code.
+  ReadToken(is, binary, &tok);
+
+  // read target_rms_ if it is available.
+  if (tok == "<TargetRms>") {
+    ReadBasicType(is, binary, &target_rms_);
+    ReadToken(is, binary, &tok);
+  }
+
+  if (tok == "<ValueSum>") {
+    // this branch is for back compatibility.  TODO: delete it
+    // after Dec 2015.
+    value_sum_.Read(is, binary);
+    ExpectToken(is, binary, "<DerivSum>");
+    deriv_sum_.Read(is, binary);
+    ExpectToken(is, binary, "<Count>");
+    ReadBasicType(is, binary, &count_);
+    ExpectToken(is, binary, ostr_end.str());
+  } else {
+    // The new format is more readable as we write values that are normalized by
+    // the count.
+    KALDI_ASSERT(tok == "<ValueAvg>");
+    value_sum_.Read(is, binary);
+    ExpectToken(is, binary, "<DerivAvg>");
+    deriv_sum_.Read(is, binary);
+    ExpectToken(is, binary, "<Count>");
+    ReadBasicType(is, binary, &count_);
+    value_sum_.Scale(count_);
+    deriv_sum_.Scale(count_);
+    ExpectToken(is, binary, ostr_end.str());
+  }
+}
 
+void NormalizeComponent::Write(std::ostream &os, bool binary) const {
+  std::ostringstream ostr_beg, ostr_end;
+  ostr_beg << "<" << Type() << ">";
+  ostr_end << "</" << Type() << ">"; 
+  WriteToken(os, binary, ostr_beg.str());
+  WriteToken(os, binary, "<Dim>");
+  WriteBasicType(os, binary, dim_);
+  WriteToken(os, binary, "<TargetRms>");
+  WriteBasicType(os, binary, target_rms_); 
+  // Write the values and derivatives in a count-normalized way, for
+  // greater readability in text form.
+  WriteToken(os, binary, "<ValueAvg>");
+  Vector<BaseFloat> temp(value_sum_);
+  if (count_ != 0.0) temp.Scale(1.0 / count_);
+  temp.Write(os, binary);
+  WriteToken(os, binary, "<DerivAvg>");
+
+  temp.Resize(deriv_sum_.Dim(), kUndefined);
+  temp.CopyFromVec(deriv_sum_);
+  if (count_ != 0.0) temp.Scale(1.0 / count_);
+  temp.Write(os, binary);
+  WriteToken(os, binary, "<Count>");
+  WriteBasicType(os, binary, count_);
+  WriteToken(os, binary, ostr_end.str());
+}
+
+std::string NormalizeComponent::Info() const {
+  std::stringstream stream;
+  KALDI_ASSERT(InputDim() == OutputDim());  // always the case
+  stream << Type() << ", dim=" << dim_ 
+         << ", target_rms=" << target_rms_;
+
+  if (count_ > 0 && value_sum_.Dim() == dim_ &&  deriv_sum_.Dim() == dim_) {
+    stream << ", count=" << std::setprecision(3) << count_
+           << std::setprecision(6);
+    Vector<double> value_avg_dbl(value_sum_);
+    Vector<BaseFloat> value_avg(value_avg_dbl);
+    value_avg.Scale(1.0 / count_);
+    stream << ", value-avg=" << SummarizeVector(value_avg);
+    Vector<double> deriv_avg_dbl(deriv_sum_);
+    Vector<BaseFloat> deriv_avg(deriv_avg_dbl);
+    deriv_avg.Scale(1.0 / count_);
+    stream << ", deriv-avg=" << SummarizeVector(deriv_avg);
+  }
+  return stream.str();
+}
+// The output y_i = target_rms * x_i / rms(x), so rms(y) = target_rms,
```

Comment:
Pegah, I think it turns out I was mistaken about this function setting the rms to a certain value, it sets the vector norm to that value.  So this makes this comment wrong, and (sorry) it also makes the choice of variable name wrong.  It should be target-norm instead of target-rms.